### PR TITLE
add session storage file

### PR DESCRIPTION
**Pull request by lucyrose93**
relates #67

@des-des feel free to merge, although I predict that you will have some helpful suggestions

I had to override the Lint pre-commit hook, as it complained that sessionStorage / location are not defined, even though they are browser features, right? 

Thanks in advance!

### CODE REVIEW

File: public/session-storage.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+(function collectAnswers () {
```
Comment (des-des) [inline public/session-storage.js:3]: At the top of this file:
```js
/* global: location, sessionStorage */
```
http://eslint.org/docs/user-guide/configuring#specifying-globals

File: public/session-storage.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    var postSession = document.getElementById('post-session-button')
+    postSession.addEventListener('click', function () {
+      sessionStorage.setItem('survey', 'post-session')
+    })
+  }
+
+  // obtain consent
+  if (location.pathname === '/consent') {
+    var yes = document.getElementById('yes')
```
Comment (des-des) [inline public/session-storage.js]: `yes` is not a good name for an id. `agree-consent` would be better as yes is not specific enough for an id name

File: public/session-storage.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+  // pre-session responses
+  if (location.pathname === '/today' ||
+    location.pathname === '/last-week' ||
+    location.pathname === '/home' ||
+    location.pathname === '/friends' ||
+    location.pathname === '/school' ||
+    location.pathname === '/play' ||
+    location.pathname === 'next-week') {
+    var veryGood = document.getElementById('very-good')
```
Comment (des-des) [inline public/session-storage.js]: again not good name for Id

File: public/session-storage.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      sessionStorage.setItem('consent', 'yes')
+    })
+
+    var no = document.getElementById('no')
+    no.addEventListener('click', function () {
+      sessionStorage.setItem('consent', 'no')
+    })
+  }
+
+  // pre-session responses
```
Comment (des-des) [inline public/session-storage.js]: Here,
```js
var path = location.pathname.slice(1)
```

File: public/session-storage.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  if (location.pathname === '/today' ||
```
Comment (des-des) [inline public/session-storage.js]:
```js
if (path === 'today' || 
   path === 'last-week'
   ...) {

}
```

File: public/session-storage.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    location.pathname === '/last-week' ||
+    location.pathname === '/home' ||
+    location.pathname === '/friends' ||
+    location.pathname === '/school' ||
+    location.pathname === '/play' ||
+    location.pathname === 'next-week') {
+    var veryGood = document.getElementById('very-good')
+    veryGood.addEventListener('click', function () {
+      sessionStorage.setItem((location.pathname).slice(1), 'very good')
```
Comment (des-des) [inline public/session-storage.js]: sessionStorage.setItem(path, 'very good')

File: public/session-storage.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    })
+
+    var good = document.getElementById('good')
```
Comment (des-des) [inline public/session-storage.js]: id name

File: public/session-storage.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    good.addEventListener('click', function () {
+      sessionStorage.setItem((location.pathname).slice(1), 'good')
```
Comment (des-des) [inline public/session-storage.js]: (location.pathname).slice(1) -> path

File: public/session-storage.js [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    })
+
+    var ok = document.getElementById('ok')
```
Comment (des-des) [inline public/session-storage.js]: same

File: public/session-storage.js [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    ok.addEventListener('click', function () {
+      sessionStorage.setItem((location.pathname).slice(1), 'ok')
```
Comment (des-des) [inline public/session-storage.js]: same

File: public/session-storage.js [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+    })
+
+    var bad = document.getElementById('bad')
```
Comment (des-des) [inline public/session-storage.js]: same

File: public/session-storage.js [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+    bad.addEventListener('click', function () {
+      sessionStorage.setItem((location.pathname).slice(1), 'bad')
```
Comment (des-des) [inline public/session-storage.js]: same

File: public/session-storage.js [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+    })
+
+    var veryBad = document.getElementById('very-bad')
```
Comment (des-des) [inline public/session-storage.js]: same

File: public/session-storage.js [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+    veryBad.addEventListener('click', function () {
+      sessionStorage.setItem((location.pathname).slice(1), 'very bad')
```
Comment (des-des) [inline public/session-storage.js]: same

File: public/session-storage.js [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+    })
+  }
+
+  // post-session responses
+  if (location.pathname === '/listen' ||
```
Comment (des-des) [inline public/session-storage.js]: as above

**Review by des-des — CHANGES_REQUESTED** [review]
Okay this looks good.

I think just clean it up a bit by pulling things you are using lots into there own variables, and giving ids more specific names.

I have not made comments at the end, but I think it should be clear

**Comment by des-des** [comment]
@lucyrose93 assign me when you want me to merge / review, sometimes I miss a notification, then will only look at the things that I am assigned to.

Also, sometimes I will assume that I do not need to look at something if I am not assigned

File: public/session-storage.js (hunk 1 above)
Comment (lucyrose93) [reply public/session-storage.js:3]: OK

File: public/session-storage.js (hunk 2 above)
Comment (lucyrose93) [inline public/session-storage.js]: There is only one "yes" option in partials

It gets reused in different contexts (hence the if statements), so I can't make the id specific like "agree-consent"

Should it be a class then?

File: public/session-storage.js (hunk 4 above)
Comment (lucyrose93) [reply public/session-storage.js]: makes sense :)

File: public/session-storage.js (hunk 5 above)
Comment (lucyrose93) [reply public/session-storage.js]: done

File: public/session-storage.js (hunk 3 above)
Comment (lucyrose93) [reply public/session-storage.js]: ☝️

File: public/session-storage.js (hunk 6 above)
Comment (lucyrose93) [reply public/session-storage.js]: :)

**Comment by lucyrose93** [comment]
@des-des thanks for the feedback ✨ 
I have incorporated all changes
apart from the id names
please see my comment regarding this above
Ta!

**Comment by lucyrose93** [comment]
@des-des I have finished making changes. 

Please merge if happy with it. 

Thanks!

**Review by des-des — APPROVED** [review]

**Pull request merged**